src/moov/trak/mdia/minf/stbl/stsd/chnl.rs

```diff
@@ -0,0 +1,396 @@
+}
+
+impl Chnl {
+    // Duplicated with slight difference to decode_body which allows
+    // Pcm box to decode the channel layout by using the channel count
+    // information from audio sample entry.
+    pub fn decode_body_with_channel_count<B: Buf>(buf: &mut B, channel_count: u16) -> Result<Self> {
+        let version_and_flags = u32::decode(buf)?;
+        let version = (version_and_flags >> 24) as u8;
+        let flags = version_and_flags & 0xFFFFFF;
+
+        if version != 0 {
+            return Err(Error::UnknownVersion(version));
+        }
+
+        if flags != 0 {
+            return Err(Error::Unsupported("chnl box with non-zero flags"));
+        }
+
+        let stream_structure = u8::decode(buf)?;
+        let channel_structured = (stream_structure & 0x01) != 0;
+        let object_structured = (stream_structure & 0x02) != 0;
+
+        let channel_structure = if channel_structured {
+            let defined_layout = u8::decode(buf)?;
+
+            if defined_layout == 0 {
+                // Explicit positions: read exactly channel_count positions
+                let mut positions = Vec::with_capacity(channel_count as usize);
+
+                for _ in 0..channel_count {
+                    let speaker_position = u8::decode(buf)?;
+
+                    if speaker_position == 126 {
+                        let azimuth = i16::decode(buf)?;
+                        let elevation = i8::decode(buf)?;
+                        positions.push(SpeakerPosition::Explicit(ExplicitSpeakerPosition {
+                            azimuth,
+                            elevation,
+                        }));
+                    } else if let Some(pos) = AudioChannelPosition::from_u8(speaker_position) {
+                        positions.push(SpeakerPosition::Standard(pos));
+                    } else {
+                        return Err(Error::Unsupported("invalid speaker position"));
+                    }
+                }
+
+                Some(ChannelStructure::ExplicitPositions { positions })
+            } else {
+                let omitted_channels_map = u64::decode(buf)?;
+                Some(ChannelStructure::DefinedLayout {
+                    layout: defined_layout,
+                    omitted_channels_map,
+                })
+            }
+        } else {
+            None
+        };
+
+        let object_count = if object_structured {
+            Some(u8::decode(buf)?)
+        } else {
+            None
+        };
+
+        Ok(Self {
+            channel_structure,
+            object_count,
+        })
+    }
+}
+
+impl Atom for Chnl {
+    const KIND: FourCC = FourCC::new(b"chnl");
+
+    fn decode_body<B: Buf>(buf: &mut B) -> Result<Self> {
+        let version_and_flags = u32::decode(buf)?;
+        let version = (version_and_flags >> 24) as u8;
+        let flags = version_and_flags & 0xFFFFFF;
+
+        if version != 0 {
+            return Err(Error::UnknownVersion(version));
```

Comment:
It looks like there is a version == 1 case for this box that isn't getting handled here.

Reply:
Yes, did not implement that for now.